apis/candidates.js

```diff
@@ -366,12 +366,30 @@ router.put('/update', [
     }
 })
 
-router.get('/:candidate/generateMessage', async function (req, res, next) {
+router.post('/:candidate/generateMessage', async function (req, res, next) {
     try {
         const candidate = req.params.candidate
+        const account = (req.body.account || '').toLowerCase()
+
+        const c = await db.Candidate.findOne({
+            smartContractAddress: config.get('blockchain.validatorAddress'),
+            candidate: candidate
+        })
+        if (!c) {
+            return res.status(406).send('This address is not a candidate')
+        }
+
         const message = '[Tomomaster ' + (new Date().toLocaleString().replace(/['"]+/g, '')) + ']' +
             ' I am the candidate ' + '[' + candidate + ']'
         const id = uuidv4()
+
+        // update id, status
+        const data = {
```

Comment:
seems lack:  `signedAddress: account`

Reply:
it will be checked in this line
`await db.Signature.findOneAndUpdate({ signedAddress: account }, data, { upsert: true, new: true })`